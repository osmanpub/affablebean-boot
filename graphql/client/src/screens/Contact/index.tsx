import React from "react";
import ContactForm from "../../components/ContactForm";
import Footer from "../../components/Footer";
import Header from "../../components/Header";
import { Match } from "../../interfaces/router";

type Props = {
  match: Match;
};

function Contact(props: Props) {
  const { match } = props;

  return (
    <div>
      <Header url={match.url} />
      <ContactForm />
      <Footer />
    </div>
  );
}

export default Contact;
