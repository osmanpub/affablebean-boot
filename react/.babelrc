 
{
  "presets": ["@babel/preset-react"],
  "env": {
    "test": {
      "plugins": [
        ["istanbul", {
          "exclude": [
            "**/*.spec.js"
          ]
        }]
      ]
    }
  },
  "plugins": ["transform-class-properties", "istanbul"]
}
