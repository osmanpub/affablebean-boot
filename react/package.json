{
  "name": "affablebean",
  "version": "0.1.0",
  "private": true,
  "eslintConfig": {
    "extends": "react-app",
    "plugins": [
      "react-hooks"
    ],
    "rules": {
      "react-hooks/rules-of-hooks": "error",
      "react-hooks/exhaustive-deps": "warn"
    }
  },
  "scripts": {
    "build": "react-scripts build",
    "coveralls": "cat coverage/lcov.info | coveralls",
    "cypress:open": "cypress open",
    "cypress:run": "cypress run",
    "cypress:verify": "cypress verify",
    "eject": "react-scripts eject",
    "report:coverage": "nyc report --reporter=html",
    "report:coverage:text": "nyc report --reporter=text",
    "start": "node_modules/react-scripts/bin/react-scripts.js start &",
    "start:java": "cd .. && sh server && cd react && node_modules/react-scripts/bin/react-scripts.js start &",
    "test": "react-scripts test"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "dependencies": {
    "@reduxjs/toolkit": "^1.2.5",
    "axios": "^0.19.2",
    "node-rest-client": "^3.1.0",
    "react": "^16.12.0",
    "react-alert": "^6.0.0",
    "react-alert-template-basic": "^1.0.0",
    "react-dom": "^16.12.0",
    "react-hook-form": "^4.9.7",
    "react-redux": "^7.2.0",
    "react-router-dom": "^5.1.2",
    "react-scripts": "3.4.0",
    "redux": "^4.0.5",
    "styled-components": "^5.0.1"
  },
  "devDependencies": {
    "@babel/core": "^7.8.4",
    "@babel/preset-react": "^7.8.3",
    "@cypress/code-coverage": "^1.12.2",
    "@testing-library/jest-dom": "^4.2.4",
    "@testing-library/react": "^9.4.0",
    "@testing-library/user-event": "^7.2.1",
    "@types/axios": "^0.14.0",
    "@types/cypress": "^1.1.3",
    "@types/jest": "^25.1.3",
    "@types/react": "^16.9.21",
    "@types/react-alert": "^5.2.0",
    "@types/react-bootstrap": "^0.32.20",
    "@types/react-dom": "^16.9.5",
    "@types/react-redux": "^7.1.7",
    "@types/react-router-dom": "^5.1.3",
    "@types/react-test-renderer": "^16.9.2",
    "@types/redux": "^3.6.0",
    "@types/redux-mock-store": "^1.0.2",
    "@types/styled-components": "^5.0.0",
    "babel-plugin-istanbul": "^6.0.0",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "coveralls": "^3.0.9",
    "cross-env": "^7.0.0",
    "cypress": "^4.0.2",
    "istanbul-lib-coverage": "^3.0.0",
    "nyc": "^15.0.0",
    "react-test-renderer": "^16.12.0",
    "redux-mock-store": "^1.5.4",
    "typescript": "^3.8.2"
  },
  "nyc": {
    "sourceMap": false,
    "instrument": false
  }
}
