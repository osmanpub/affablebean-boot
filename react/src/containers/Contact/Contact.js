import React, { Component } from "react";
import { connect } from "react-redux";
import ContactForm from "../../components/ContactForm";
import Footer from "../../components/Footer";
import Header from "../../components/Header";
import { fetchSubjectsIfNeeded } from "../../net/subjects";

export class Contact extends Component {
  componentDidMount() {
    const { dispatch } = this.props;
    dispatch(fetchSubjectsIfNeeded());
  }

  render() {
    const { cart, match, subjects } = this.props;
    const { items } = subjects;

    if (items.length === 0) {
      return null;
    }

    return (
      <div>
        <Header cart={cart} url={match.url} />
        <ContactForm subjects={subjects} />
        <Footer />
      </div>
    );
  }
}

const mapStateToProps = state => {
  const { cart, subjects } = state;

  return {
    cart,
    subjects
  };
};

export default connect(mapStateToProps)(Contact);
