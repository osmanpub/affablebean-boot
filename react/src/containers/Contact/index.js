import Confirmation from "./Contact";

export default Confirmation;
