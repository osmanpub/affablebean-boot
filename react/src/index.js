import React from "react";
import { render } from "react-dom";
import { configureStore } from "redux-starter-kit";
import { Provider } from "react-redux";
import { BrowserRouter as Router, Route } from "react-router-dom";
import reducer from "./reducers";
import App from "./containers/App";
import CategoryProducts from "./containers/CategoryProducts";
import * as serviceWorker from "./serviceWorker";

const store = configureStore({
  reducer
});

render(
  // <React.StrictMode>
  <Provider store={store}>
    <Router>
      <Route path="/" exact component={App} />
      <Route path="/category/:id" component={CategoryProducts} />
    </Router>
  </Provider>,
  // </React.StrictMode>,
  document.getElementById("root")
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
serviceWorker.unregister();
