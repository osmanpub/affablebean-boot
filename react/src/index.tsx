import { configureStore } from "@reduxjs/toolkit";
import React, { lazy, Suspense } from "react";
import { render } from "react-dom";
import { Provider } from "react-redux";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import ErrorBoundary from "./components/ErrorBoundary";
import rootReducer from "./redux";
import * as serviceWorker from "./serviceWorker";
const Cart = lazy(() => import("./containers/Cart"));
const CategoryProducts = lazy(() => import("./containers/CategoryProducts"));
const Checkout = lazy(() => import("./containers/Checkout"));
const Contact = lazy(() => import("./containers/Contact"));
const Home = lazy(() => import("./containers/Home"));
const Privacy = lazy(() => import("./containers/Privacy"));

const store = configureStore({
  reducer: rootReducer
});

render(
  <React.StrictMode>
    <ErrorBoundary>
      <Provider store={store}>
        <Router>
          <Suspense fallback={<div>Loading...</div>}>
            <Switch>
              <Route path="/" exact component={Home} />
              <Route path="/category/:id" component={CategoryProducts} />
              <Route path="/checkout" component={Checkout} />
              <Route path="/contact" component={Contact} />
              <Route path="/privacy" component={Privacy} />
              <Route path="/viewCart/:clear" component={Cart} />
            </Switch>
          </Suspense>
        </Router>
      </Provider>
    </ErrorBoundary>
  </React.StrictMode>,
  document.getElementById("root")
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
serviceWorker.unregister();
