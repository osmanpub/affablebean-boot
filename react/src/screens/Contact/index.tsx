import React, { useEffect } from "react";
import { connect, useDispatch } from "react-redux";
import ContactForm from "../../components/ContactForm";
import Footer from "../../components/Footer";
import Header from "../../components/Header";
import { Match } from "../../interfaces/router";
import { fetchSubjectsIfNeeded } from "../../net/subjects";
import { RootState } from "../../redux";

type Props = {
  match: Match;
  subjects: any;
};

function Contact(props: Props) {
  const { match, subjects } = props;
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchSubjectsIfNeeded());
  }, [dispatch, subjects]);

  const { items } = subjects;

  if (items.length === 0) {
    return null;
  }

  return (
    <div>
      <Header url={match.url} />
      <ContactForm />
      <Footer />
    </div>
  );
}

const mapStateToProps = (state: RootState) => ({
  subjects: state.subjects
});

export default connect(mapStateToProps)(Contact);
