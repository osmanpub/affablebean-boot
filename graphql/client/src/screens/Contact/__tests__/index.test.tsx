import { act, fireEvent, render } from "@testing-library/react";
import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import Contact from "..";

const mockStore = configureStore([thunk]);

const initialState = {
  cart: {
    didInvalidate: false,
    isFetching: false,
    items: [],
    numberOfItems: 0,
    subtotal: 0,
  },
  subjects: {
    didInvalidate: false,
    isFetching: false,
    items: [
      {
        id: 1,
        name: "Brands or product",
      },
      {
        id: 2,
        name: "Investor relations",
      },
      {
        id: 3,
        name: "Sustainability",
      },
      {
        id: 4,
        name: "The Company",
      },
      {
        id: 5,
        name: "Media enquiry",
      },
      {
        id: 6,
        name: "Website feedback",
      },
      {
        id: 7,
        name: "Other",
      },
    ],
  },
  ui: {
    formErrors: [],
  },
};

const store = mockStore(initialState);

const contact = (
  <Router>
    <Provider store={store}>
      <Contact
        match={{
          path: "/contact",
          url: "/contact",
          isExact: true,
          params: {},
        }}
      />
    </Provider>
  </Router>
);

describe("<Contact />", () => {
  it("renders correctly", () => {
    const component = renderer.create(contact).toJSON();
    // @ts-ignore
    expect(component).toMatchSnapshot();
  });

  it("show smaller navigatioal logo, on left", () => {
    const { getByAltText } = render(contact);
    const logo = getByAltText(/Affable Bean logo/i);
    // @ts-ignore
    expect(logo).toBeInTheDocument();
  });

  it("show larger logo, on right", () => {
    const { getByAltText } = render(contact);
    const logo = getByAltText(/the affable bean/i);
    // @ts-ignore
    expect(logo).toBeInTheDocument();
  });

  it("don't show view cart link", () => {
    const { getByText } = render(contact);
    const link = getByText(/view cart/i);
    // @ts-ignore
    expect(link).not.toBeVisible();
  });

  it("don't show cart header total", () => {
    const { getByTestId } = render(contact);
    const cart = getByTestId(/cart-hdr-total/i);
    // @ts-ignore
    expect(cart).not.toBeVisible();
  });

  it("show intro", () => {
    const { getByTestId } = render(contact);
    const intro = getByTestId(/contact-intro/i);
    // @ts-ignore
    expect(intro).toBeInTheDocument();
  });

  it("show contact", () => {
    const { getByTestId } = render(contact);
    const intro = getByTestId(/footer-contact/i);
    // @ts-ignore
    expect(intro).toBeInTheDocument();
  });

  it("show privacy", () => {
    const { getByTestId } = render(contact);
    const privacy = getByTestId(/footer-privacy/i);
    // @ts-ignore
    expect(privacy).toBeInTheDocument();
  });
});

describe("<Data Entry />", () => {
  it("renders correctly", () => {
    const component = renderer.create(contact).toJSON();
    // @ts-ignore
    expect(component).toMatchSnapshot();
  });

  it("show intro", () => {
    const { getByTestId } = render(contact);
    const intro = getByTestId(/contact-intro/i);
    // @ts-ignore
    expect(intro).toBeInTheDocument();
  });

  it("show form", () => {
    const { getByRole } = render(contact);
    const form = getByRole(/form/i);
    // @ts-ignore
    expect(form).toBeInTheDocument();
  });

  it("show select box", () => {
    const { getByRole } = render(contact);
    const input = getByRole(/listbox/i);
    // @ts-ignore
    expect(input).toBeInTheDocument();
  });

  it("show 3 input fields", () => {
    const { getAllByRole } = render(contact);
    const inputs = getAllByRole(/textbox/i);
    // @ts-ignore
    expect(inputs).toHaveLength(3);
  });

  it("set name to 'osman'", () => {
    const { getByTestId } = render(contact);
    const input = getByTestId(/contact-name/i);

    act(() => {
      fireEvent.change(input, { target: { value: "osman" } });
    });

    // @ts-ignore
    expect(input.value).toBe("osman");
  });

  it("set email to 'osman@gmail.com'", () => {
    const { getByTestId } = render(contact);
    const input = getByTestId(/contact-email/i);

    act(() => {
      fireEvent.change(input, { target: { value: "osman@gmail.com" } });
    });

    // @ts-ignore
    expect(input.value).toBe("osman@gmail.com");
  });

  it("set message to 'hello there!'", () => {
    const { getByTestId } = render(contact);
    const input = getByTestId(/contact-msg/i);

    act(() => {
      fireEvent.change(input, { target: { value: "hello there!" } });
    });

    // @ts-ignore
    expect(input.value).toBe("hello there!");
  });
});
