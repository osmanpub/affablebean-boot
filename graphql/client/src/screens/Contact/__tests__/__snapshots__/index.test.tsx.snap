// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact /> renders correctly 1`] = `
<div>
  <div
    className="sc-AxgMl kehYEp"
  >
    <div
      className="sc-AxheI jbKZFY"
    >
      <div
        className="sc-AxiKw byMOZe"
      >
        <div
          className="sc-AxhCb kneLnw headerWidget"
        >
          <a
            className="bubble"
            data-cy="proceed-checkout"
            href="/checkout"
            onClick={[Function]}
          >
            proceed to checkout ➟
          </a>
        </div>
        <div
          className="sc-AxhUy dXnRMF headerWidget"
        >
          <span
            className="horizontalMargin"
            data-cy="cart-hdr-total"
            data-testid="cart-hdr-total"
          >
            <img
              alt="shopping cart icon"
              src="/static/img/cart.gif"
            />
             
            0 items
          </span>
          <a
            className="bubble"
            data-cy="cart-hdr-view-cart"
            data-testid="cart-hdr-view-cart"
            href="/viewCart/false"
            onClick={[Function]}
          >
            view cart
          </a>
        </div>
      </div>
      <div
        className="sc-Axmtr cYyhem"
      >
        <a
          data-cy="home-logo"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="Affable Bean logo"
            src="/static/img/logo.png"
          />
        </a>
      </div>
      <img
        alt="the affable bean"
        id="logoText"
        src="/static/img/logoText.png"
      />
    </div>
  </div>
  <div
    className="singleColumn"
  >
    <div>
      <h2
        data-cy="contact-intro"
        data-testid="contact-intro"
      >
        Contact Us Form
      </h2>
      <p>
        You can use this form for any comments or questions about our company or brands.
        <br />
        For general enquiries please call toll free on 1-800-BEANS-R-US
      </p>
    </div>
    <div>
      <p>
        (* Asterisks indicate required field)
      </p>
    </div>
    <form
      className="form-horizontal"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="col-sm-2 control-label"
          htmlFor="subject"
        >
          subject
        </label>
        <div
          className="col-sm-10"
        >
          <select
            className="form-control"
            data-cy="contact-subject"
            data-testid="contact-subject"
            name="subject"
          >
            <option
              value={1}
            >
              Brands or product
            </option>
            <option
              value={2}
            >
              Investor relations
            </option>
            <option
              value={3}
            >
              Sustainability
            </option>
            <option
              value={4}
            >
              The Company
            </option>
            <option
              value={5}
            >
              Media enquiry
            </option>
            <option
              value={6}
            >
              Website feedback
            </option>
            <option
              value={7}
            >
              Other
            </option>
          </select>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="col-sm-2 control-label"
          htmlFor="name"
        >
          name
        </label>
        <div
          className="col-sm-10"
        >
          <input
            className="form-control"
            data-cy="contact-name"
            data-testid="contact-name"
            name="name"
            placeholder="At least 3 chars and no more than 64 chars"
            size={31}
            type="text"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="col-sm-2 control-label"
          htmlFor="email"
        >
          email
        </label>
        <div
          className="col-sm-10"
        >
          <input
            className="form-control"
            data-cy="contact-email"
            data-testid="contact-email"
            name="email"
            placeholder="At least 8 chars and no more than 32 chars"
            size={31}
            type="email"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="col-sm-2 control-label"
          htmlFor="msg"
        >
          message
        </label>
        <div
          className="col-sm-10"
        >
          <textarea
            className="form-control"
            cols={25}
            data-cy="contact-msg"
            data-testid="contact-msg"
            name="msg"
            placeholder="At least 8 chars and no more than 1024 chars"
            rows={5}
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="col-sm-offset-2 col-sm-10"
        >
          <button
            className="btn btn-primary"
            data-cy="contact-submit"
            data-testid="contact-submit"
            type="submit"
          >
            send feedback
          </button>
        </div>
      </div>
    </form>
    <div>
      <br />
      <h3>
        Privacy
      </h3>
      <p>
        This website is owned and controlled by:
        <br />
        <br />
         The Affable Bean Company Limited 
        <br />
        Registered office: AffableBean House, Roast Drive, 
        <br />
         Pothead, POT123, BeanieLand
      </p>
      <p>
        (hereinafter referred to as "AffableBean")
      </p>
      <p>
        AffableBean is responsible for the processing of any information collected by or on behalf of AffableBean on or via this website. All such information will be processed in accordance with our 
        <a
          href="/privacy"
          onClick={[Function]}
        >
          privacy policy
        </a>
        .
      </p>
    </div>
  </div>
  <div
    className="sc-AxjAm jZHPdI"
  >
    <br />
    <span
      className="sc-AxirZ bSLrDO"
    >
      <a
        data-cy="privacy"
        data-testid="footer-privacy"
        href="/privacy"
        onClick={[Function]}
      >
        <button
          className="btn btn-link"
        >
          Privacy
        </button>
      </a>
      <a
        data-cy="contact"
        data-testid="footer-contact"
        href="/contact"
        onClick={[Function]}
      >
        <button
          className="btn btn-link"
        >
          Contact
        </button>
      </a>
       © 2020 the affable bean company
    </span>
  </div>
</div>
`;

exports[`<Data Entry /> renders correctly 1`] = `
<div>
  <div
    className="sc-AxgMl kehYEp"
  >
    <div
      className="sc-AxheI jbKZFY"
    >
      <div
        className="sc-AxiKw byMOZe"
      >
        <div
          className="sc-AxhCb kneLnw headerWidget"
        >
          <a
            className="bubble"
            data-cy="proceed-checkout"
            href="/checkout"
            onClick={[Function]}
          >
            proceed to checkout ➟
          </a>
        </div>
        <div
          className="sc-AxhUy dXnRMF headerWidget"
        >
          <span
            className="horizontalMargin"
            data-cy="cart-hdr-total"
            data-testid="cart-hdr-total"
          >
            <img
              alt="shopping cart icon"
              src="/static/img/cart.gif"
            />
             
            0 items
          </span>
          <a
            className="bubble"
            data-cy="cart-hdr-view-cart"
            data-testid="cart-hdr-view-cart"
            href="/viewCart/false"
            onClick={[Function]}
          >
            view cart
          </a>
        </div>
      </div>
      <div
        className="sc-Axmtr cYyhem"
      >
        <a
          data-cy="home-logo"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="Affable Bean logo"
            src="/static/img/logo.png"
          />
        </a>
      </div>
      <img
        alt="the affable bean"
        id="logoText"
        src="/static/img/logoText.png"
      />
    </div>
  </div>
  <div
    className="singleColumn"
  >
    <div>
      <h2
        data-cy="contact-intro"
        data-testid="contact-intro"
      >
        Contact Us Form
      </h2>
      <p>
        You can use this form for any comments or questions about our company or brands.
        <br />
        For general enquiries please call toll free on 1-800-BEANS-R-US
      </p>
    </div>
    <div>
      <p>
        (* Asterisks indicate required field)
      </p>
    </div>
    <form
      className="form-horizontal"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="col-sm-2 control-label"
          htmlFor="subject"
        >
          subject
        </label>
        <div
          className="col-sm-10"
        >
          <select
            className="form-control"
            data-cy="contact-subject"
            data-testid="contact-subject"
            name="subject"
          >
            <option
              value={1}
            >
              Brands or product
            </option>
            <option
              value={2}
            >
              Investor relations
            </option>
            <option
              value={3}
            >
              Sustainability
            </option>
            <option
              value={4}
            >
              The Company
            </option>
            <option
              value={5}
            >
              Media enquiry
            </option>
            <option
              value={6}
            >
              Website feedback
            </option>
            <option
              value={7}
            >
              Other
            </option>
          </select>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="col-sm-2 control-label"
          htmlFor="name"
        >
          name
        </label>
        <div
          className="col-sm-10"
        >
          <input
            className="form-control"
            data-cy="contact-name"
            data-testid="contact-name"
            name="name"
            placeholder="At least 3 chars and no more than 64 chars"
            size={31}
            type="text"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="col-sm-2 control-label"
          htmlFor="email"
        >
          email
        </label>
        <div
          className="col-sm-10"
        >
          <input
            className="form-control"
            data-cy="contact-email"
            data-testid="contact-email"
            name="email"
            placeholder="At least 8 chars and no more than 32 chars"
            size={31}
            type="email"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="col-sm-2 control-label"
          htmlFor="msg"
        >
          message
        </label>
        <div
          className="col-sm-10"
        >
          <textarea
            className="form-control"
            cols={25}
            data-cy="contact-msg"
            data-testid="contact-msg"
            name="msg"
            placeholder="At least 8 chars and no more than 1024 chars"
            rows={5}
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="col-sm-offset-2 col-sm-10"
        >
          <button
            className="btn btn-primary"
            data-cy="contact-submit"
            data-testid="contact-submit"
            type="submit"
          >
            send feedback
          </button>
        </div>
      </div>
    </form>
    <div>
      <br />
      <h3>
        Privacy
      </h3>
      <p>
        This website is owned and controlled by:
        <br />
        <br />
         The Affable Bean Company Limited 
        <br />
        Registered office: AffableBean House, Roast Drive, 
        <br />
         Pothead, POT123, BeanieLand
      </p>
      <p>
        (hereinafter referred to as "AffableBean")
      </p>
      <p>
        AffableBean is responsible for the processing of any information collected by or on behalf of AffableBean on or via this website. All such information will be processed in accordance with our 
        <a
          href="/privacy"
          onClick={[Function]}
        >
          privacy policy
        </a>
        .
      </p>
    </div>
  </div>
  <div
    className="sc-AxjAm jZHPdI"
  >
    <br />
    <span
      className="sc-AxirZ bSLrDO"
    >
      <a
        data-cy="privacy"
        data-testid="footer-privacy"
        href="/privacy"
        onClick={[Function]}
      >
        <button
          className="btn btn-link"
        >
          Privacy
        </button>
      </a>
      <a
        data-cy="contact"
        data-testid="footer-contact"
        href="/contact"
        onClick={[Function]}
      >
        <button
          className="btn btn-link"
        >
          Contact
        </button>
      </a>
       © 2020 the affable bean company
    </span>
  </div>
</div>
`;
